packages/Dendrogram/src/utils/tree-helper.ts

```diff
@@ -460,22 +463,36 @@ export class TreeHelper implements ITreeHelper {
     return treeRoot;
   }
 
-  async calcDistanceMatrix(df: DG.DataFrame, colNames: string[], method: DistanceMetric = DistanceMetric.Euclidean) {
+  async calcDistanceMatrix(
+    df: DG.DataFrame, colNames: string[], method: DistanceMetric = DistanceMetric.Euclidean
+  ) {
+    // Output distance matrix. reusing it saves a lot of memory
     let out: DistanceMatrix | null = null;
+
+    const distanceMatrixWorker = new DistanceMatrixWorker();
     const columns = colNames.map((name) => df.getCol(name));
+    const hammingColumns: string[] = [];
     for (const col of columns) {
       let values: Float32Array;
-      if (col.type === DG.TYPE.FLOAT || col.type === DG.TYPE.INT)
-        values = await getDistanceMatrixForNumerics(col.toList() as unknown as Float32Array);
-      else if (col.semType ==='Macromolecule')
-        values = await getDistanceMatrixForSequences(col.toList() as unknown as string[]);
-      else throw new TypeError('Unsupported column type');
+      if (col.type === DG.TYPE.FLOAT || col.type === DG.TYPE.INT) {
```

Comment:
We also have DG.TYPE.QNUM; perhaps a helper method isNumerical is needed somewhere